### PR TITLE
Factored out EditorWithToolbar class.

### PR DESCRIPTION
**Pull request by toolness**
This just combines the editor with the toolbar and changes the toolbar class to not display the Publish button if no `startPublish` callback has been set.

This refactoring provides a better separation of concerns between editing and publishing, and allows the `EditorWithToolbar` widget to be embedded in a webpage if no publish functionality is needed.

### CODE REVIEW

File: js/fc/ui/editor-with-toolbar.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+define([
+  "jquery",
+  "./two-paned-editor",
+  "./editor-toolbar"
+], function($, TwoPanedEditor, EditorToolbar) {
+  return function EditorWithToolbar(options) {
```
Comment (stenington) [inline js/fc/ui/editor-with-toolbar.js]: My eye keeps missing the difference between `EditorToolbar` and `EditorWithToolbar`, which leaves me _really_ confused. Maybe `EditorToolbar` can just be `Toolbar`?

**Comment by stenington** [comment]
The idea of an `XWithY` class feels weird to me at first, but maybe it's just a naming thing? If the editor really needs a toolbar to function (such that they're more useful prepackaged in a single class) then maybe it should just be called Editor, and it should come with the toolbar that you can opt to override or remove?

File: js/fc/ui/editor-with-toolbar.js (hunk 1 above)
Comment (toolness) [inline js/fc/ui/editor-with-toolbar.js]: That's not a bad idea ... If we did that, would `EditorWithToolbar`'s name stay the same, or would it change too?

Comment (stenington) [inline js/fc/ui/editor-with-toolbar.js]: I would recommend that `EditorWithToolbar` becomes `Editor`, and `EditorToolbar` becomes `Toolbar`, I think.

**Comment by toolness** [comment]
Ok, after talking to stenington on IRC it sounds like the best way to go is the following renaming:

```
TwoPanedEditor -> EditorPanes
EditorWithToolbar -> Editor
```

`EditorToolbar` doesn't get renamed.

**Pull request merged**